cmake/apple/CMakeLists.txt

```diff
@@ -19,8 +19,22 @@ CONFIGURE_FILE("lmms.plist.in"          "${CMAKE_BINARY_DIR}/Info.plist")
 CONFIGURE_FILE("install_apple.sh.in"    "${CMAKE_BINARY_DIR}/install_apple.sh" @ONLY)
 CONFIGURE_FILE("package_apple.json.in"  "${CMAKE_BINARY_DIR}/package_apple.json" @ONLY)
 
+IF(CMAKE_APPLE_SILICON_PROCESSOR)
+	# Target arch is provided to cmake (M-series only)
+	SET(DMG_ARCH "-${CMAKE_APPLE_SILICON_PROCESSOR}")
+ELSEIF(CMAKE_OSX_ARCHITECTURES)
+	SET(DMG_ARCH "-${CMAKE_OSX_ARCHITECTURES}")
```

Comment:
Hmm... weird.  While writing #7278  I think I misunderstood the behavior testing locally.   You're right, the host architecture said "Intel" but Rosetta 2 (deliberately) isn't installed in my VM so I'm not sure what this was actually doing.

To that point, it may make sense to bring it back in the future... combined with running homebrew with `-arch x86_64` this may greatly simplify the yaml file in the future when GitHub deprecates Intel runners.  For now, I agree it offers no benefit.

WIth regards to `CMAKE_OSX_ARCHITECTURES` as a side note, it's technically not needed, but removes ambiguity.

> I think it would look a little cleaner to add the "-" separator in the set(DMG_FILE) command below, rather than in each separate case here.

Agreed.  This was a copy/paste from #7278 added it conditionally, but the logic has since changed.